lib/Common/Core/SysInfo.h

```diff
@@ -62,7 +63,11 @@ class AutoSystemInfo : public SYSTEM_INFO
 #ifdef _WIN32
     static HMODULE GetCRTHandle();
 #endif
+#if defined(__APPLE__) && defined(_M_ARM64)
+    static DWORD const PageSize = 16384;
```

Comment:
Nit: are there any consequences to hardcoding the page size? I think base class should contain the page size (this might be out of scope of this review).

Reply:
I was staying consistent with how it was before, it may not be ideal but it's not a change.